src/analysis/normed_space/add_torsor.lean

```diff
@@ -27,35 +27,37 @@ structure and require the distance to be the same as results from the
 norm (which in fact implies the distance yields a metric space, but
 bundling just the distance and using an instance for the metric space
 results in type class problems). -/
-class normed_add_torsor (V : Type u) (P : Type v) [normed_group V] [metric_space P]
+class normed_add_torsor (V : out_param $ Type u) (P : Type v)
+  [out_param $ normed_group V] [metric_space P]
   extends add_torsor V P :=
 (dist_eq_norm' : ∀ (x y : P), dist x y = ∥(x -ᵥ y : V)∥)
 end prio
 
+variables (V : Type u) {P : Type v} [normed_group V] [metric_space P] [normed_add_torsor V P]
+include V
+
 /-- The distance equals the norm of subtracting two points. This lemma
 is needed to make V an explicit rather than implicit argument. -/
-lemma add_torsor.dist_eq_norm (V : Type u) {P : Type v} [normed_group V] [metric_space P]
-    [normed_add_torsor V P] (x y : P) :
-  dist x y = ∥(x -ᵥ y : V)∥ :=
+lemma dist_eq_norm_vsub (x y : P) :
+  dist x y = ∥(x -ᵥ y)∥ :=
 normed_add_torsor.dist_eq_norm' x y
```

Comment:
I would hope that `V` can become an implicit argument to this lemma now, is there some reason it needs to stay explicit?

Reply:
I don't understand the reason but `rw [dist_eq_norm_vsub]` fails otherwise.

Reply:
The weird thing is that `rw [dist_eq_norm_vsub _, ...]` also works, but no variations on the order or making it `{{double-bracketed}}` seem to help.

Could you note the reason that `V` is explicit in a comment?

Reply:
I've expanded the doc string to say that `rw` fails otherwise.